File: library/src/main/java/renetik/android/java/lang/CSStringBuilderWriter.java
package renetik.android.java.lang;

import java.io.Serializable;
import java.io.Writer;

/**
 * Created by Rene Dohan on 07/05/15.
 */
public class CSStringBuilderWriter extends Writer implements Serializable {

    private final StringBuilder builder;

    public CSStringBuilderWriter() {
        this.builder = new StringBuilder();
    }

    public CSStringBuilderWriter(int capacity) {
        this.builder = new StringBuilder(capacity);
    }

    public CSStringBuilderWriter(StringBuilder builder) {
        this.builder = (builder != null ? builder : new StringBuilder());
    }

    public Writer append(char value) {
        builder.append(value);
        return this;
    }

    public Writer append(CharSequence value) {
        builder.append(value);
        return this;
    }

    public Writer append(CharSequence value, int start, int end) {
        builder.append(value, start, end);
        return this;
    }

    public void close() {
    }

    public void flush() {
    }

    public void write(String value) {
        if (value != null) builder.append(value);
    }

    public void write(char[] value, int offset, int length) {
        if (value != null) builder.append(value, offset, length);
    }

    public StringBuilder getBuilder() {
        return builder;
    }

    public String toString() {
        return builder.toString();
    }
}

File: library/base/src/main/java/renetik/android/java/common/CSStringBuilderWriter.kt
package renetik.android.java.common

import java.io.Serializable
import java.io.Writer

class CSStringBuilderWriter : Writer, Serializable {

    private val builder: StringBuilder

    constructor() {
        this.builder = StringBuilder()
    }

    constructor(capacity: Int) {
        this.builder = StringBuilder(capacity)
    }

    constructor(builder: StringBuilder?) {
        this.builder = builder ?: StringBuilder()
    }

    override fun append(value: Char): Writer {
        builder.append(value)
        return this
    }

    override fun append(value: CharSequence?): Writer {
        builder.append(value)
        return this
    }

    override fun append(value: CharSequence?, start: Int, end: Int): Writer {
        builder.append(value, start, end)
        return this
    }

    override fun close() {}

    override fun flush() {}

    override fun write(value: String?) {
        if (value != null) builder.append(value)
    }

    override fun write(value: CharArray?, offset: Int, length: Int) {
        if (value != null) builder.append(value, offset, length)
    }

    override fun toString(): String {
        return builder.toString()
    }
}

